### ISSUE TITLE
[Feature request] Fallback to sequential downloading if cURL is not available ?

### ISSUE THREAD

**Issue by Phyks** [stefangabos/Zebra_cURL#14]
Hi,

What do you think about falling back to a sequential downloading (using `file_get_contents` for example) if the `cURL` PHP extension is not available ?

I don't think this would be a big overhead and high number of lines of code to write, and it could provide a library that works on every system, without failing.

Thanks

**Comment by Phyks** [comment]
What do you think about it? Do you want a PR? Thanks

**Comment by Chienle** [comment]
Don't worry Phyks, 
I think your question is unclear, to fix that issue can you please explain or paste you code into here, we will investigation.

**Comment by stefangabos** [comment]
The question is clear to me but this *is a curl wrapper* so it doesn't really make sense to fallback to `file_get_contents` as it's not what it is supposed to do, not to mention that it requires quite a lot of work,,,

**Comment by Phyks** [comment]
Sure, I get the point about the `curl` wrapper. The idea of this issue was to offer a single API for querying multiple URL endpoints in parallel. If this is not something you find worth it, maybe just adding some method to `Zebra_cURL` to check `curl` extension is indeed available could be a useful extension in my opinion.

**Comment by stefangabos** [comment]
*querying multiple URL endpoints in parallel* is only possible when using `curl` - using `file_get_contents` would allow only for asynchronous access...

`Zebra_cURL` does check if `curl` lib is loaded and [triggers a fatal error if not](https://github.com/stefangabos/Zebra_cURL/blob/master/Zebra_cURL.php#L260)

**Comment by Phyks** [comment]
Ok. Sorry, it's been a while since I last used `Zebra_cURL`. This was some idea I had at the time, and submitted an issue. Feel free to close if invalid / wontfix :)
